clarity/src/vm/ast/definition_sorter/tests.rs

```diff
@@ -86,238 +110,312 @@ fn should_succeed_sorting_contract_case_2(#[case] version: ClarityVersion) {
+    #[case] version: ClarityVersion,
+    #[case] epoch: StacksEpochId,
+) {
     let contract = r#"
         (define-private (foo (x int)) (+ (bar x) x))
         (define-private (bar (x int)) (map-insert kv-store { foo: (foo 1) } { bar: 3 }))
         (define-map kv-store { foo: int } { bar: int })
     "#;
 
-    let err = run_scoped_parsing_helper(contract, version).unwrap_err();
+    let err = run_scoped_parsing_helper(contract, version, epoch).unwrap_err();
     assert!(match err.err {
         ParseErrors::CircularReference(_) => true,
         _ => false,
     })
 }
 
 #[apply(test_clarity_versions_definition_sorter)]
-fn should_raise_dependency_cycle_case_fetch_contract_entry(#[case] version: ClarityVersion) {
+fn should_raise_dependency_cycle_case_fetch_contract_entry(
+    #[case] version: ClarityVersion,
+    #[case] epoch: StacksEpochId,
+) {
     let contract = r#"
         (define-private (foo (x int)) (+ (bar x) x))
         (define-private (bar (x int)) (map-get? kv-store { foo: (foo 1) })) 
     "#;
 
-    let err = run_scoped_parsing_helper(contract, version).unwrap_err();
+    let err = run_scoped_parsing_helper(contract, version, epoch).unwrap_err();
     assert!(match err.err {
         ParseErrors::CircularReference(_) => true,
         _ => false,
     })
 }
 
 #[apply(test_clarity_versions_definition_sorter)]
-fn should_not_build_cycle_within_defined_args_types(#[case] version: ClarityVersion) {
+fn should_not_build_cycle_within_defined_args_types(
+    #[case] version: ClarityVersion,
+    #[case] epoch: StacksEpochId,
+) {
     let contract = r#"
         (define-private (function-1 (function-2 int)) (+ 1 2))
         (define-private (function-2 (function-1 int)) (+ 1 2))
     "#;
 
-    run_scoped_parsing_helper(contract, version).unwrap();
+    run_scoped_parsing_helper(contract, version, epoch).unwrap();
 }
 
 #[apply(test_clarity_versions_definition_sorter)]
-fn should_reorder_traits_references(#[case] version: ClarityVersion) {
+fn should_reorder_traits_references(#[case] version: ClarityVersion, #[case] epoch: StacksEpochId) {
     let contract = r#"
         (define-private (foo (function-2 <trait-a>)) (+ 1 2))
         (define-trait trait-a ((get-a () (response uint uint))))
     "#;
 
-    run_scoped_parsing_helper(contract, version).unwrap();
+    run_scoped_parsing_helper(contract, version, epoch).unwrap();
 }
 
 #[apply(test_clarity_versions_definition_sorter)]
-fn should_not_conflict_with_atoms_from_trait_definitions(#[case] version: ClarityVersion) {
+fn should_not_conflict_with_atoms_from_trait_definitions(
+    #[case] version: ClarityVersion,
+    #[case] epoch: StacksEpochId,
+) {
     let contract = r#"
         (define-trait foo ((bar (int) (int))))
         (define-trait bar ((foo (int) (int))))
     "#;
 
-    run_scoped_parsing_helper(contract, version).unwrap();
+    run_scoped_parsing_helper(contract, version, epoch).unwrap();
+}
+
+#[apply(test_clarity_versions_definition_sorter)]
```

Comment:
just want to check whether there should be tests for other calls to `filter_comments` (for example, you called it in `probe_for_dependencies_in_define_args`). Essentially, does omitting any of the other calls to `filter_comments` lead to an error?

Reply:
Yes, I will add more tests.

Reply:
Addressed in 06e546951.